File: Codewars/Python/5 kyu/Scramblies.py
from codewars_test import test_framework as Test


def scramble(s1, s2):
    temp_word = s1
    for char in s2:
        if char in temp_word:
            temp_word = temp_word.replace(char,'',1)
        else:
            return False
    return True



# Test.assert_equals(scramble('rkqodlw', 'world'),  True)
# Test.assert_equals(scramble('cedewaraaossoqqyt', 'codewars'), True)
# Test.assert_equals(scramble('katas', 'steak'), False)
# Test.assert_equals(scramble('scriptjava', 'javascript'), True)
# Test.assert_equals(scramble('scriptingjava', 'javascript'), True)

File: Codewars/JS/5ku/Scramblies.js
function scramble(str1, str2) {
    for (let x of str2) {
        if (count(str2, x) > count(str1, x)) {
            return false;
        }
    }
    return true;
}

function count(arr, char) {
    let reg = new RegExp(char, 'g'),
        count = (arr.match(reg) || []).length;
    return count;
}

console.log(scramble('rkqodlw', 'world'), true);
console.log(scramble('rkqodlw', '1world'), false);
console.log(scramble('cedewaraaossoqqyt', 'codewars'), true);
console.log(scramble('katas', 'steak'), false);
console.log(scramble('scriptjava', 'javascript'), true);
console.log(scramble('scriptingjava', 'javascript'), true);
console.log(scramble('scriptsjava', 'javascripts'), true);
console.log(scramble('jscripts', 'javascript'), false);
console.log(scramble('aabbcamaomsccdd', 'commas'), true);
